client/auth.ts

```diff
@@ -1,5 +1,23 @@
+/** @link See https://authjs.dev/getting-started/migrating-to-v5#configuration-file */
+
 import NextAuth from "next-auth";
 export const { handlers, signIn, signOut, auth } = NextAuth({
   debug: true,
-  providers: []
+  secret: process.env.NEXTAUTH_SECRET,
+  providers: [
+    {
+      id: "a12n-server", // signIn("my-provider") and will be part of the callback URL
+      name: "a12n-server", // optional, theoretically used on the default login page as the button text.
+      type: "oauth",
+      issuer: process.env.NEXTAUTH_INTERNAL_URL + `/.well-known/oauth-authorization-server`,
```

Comment:
Based on the docs, you don't need the full well-known url, as it will be discovered

```suggestion
      issuer: process.env.NEXTAUTH_INTERNAL_URL,
```